### PR TITLE
chore: 🤖 clean up build scripts to run in parallel

### PR DESCRIPTION
**Pull request by maniator**
To be merged after https://github.com/aws-observability/aws-rum-web/pull/148

See diff here: https://github.com/maniator/aws-rum-web/compare/split-out-testcafe...maniator:build-async

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,28 +17,31 @@
         "url": "https://github.com/aws-observability/aws-rum-web.git"
     },
     "scripts": {
-        "server": "npm run build:schemas && webpack-dev-server --config webpack/webpack.dev.js",
-        "test": "npm run build:webpack:dev && jest -c jest.unit.config.js",
-        "release": "npm run build:webpack:prod && npm run build:es && npm run build:cjs",
+        "server": "webpack-dev-server --config webpack/webpack.dev.js",
```
Comment (adebayor123) [inline package.json:20]: Is `build:schemas` triggered here, or is it intentionally removed?

File: package.json [hunk 2]
```diff
@@ -17,28 +17,31 @@
...
+        "start": "npm run server",
+        "pretest": "npm run build:schemas",
```
Comment (adebayor123) [inline package.json:22]: In the original, we run `build:webpack:dev` then run the tests. Here, it seems that we run `build:schemas` instead; is this intentional? Could you explain the change?

Comment (maniator) [reply package.json:22]: We don't need to build we pack before running the tests, but the schemas need to be there

File: package.json (hunk 1 above)
Comment (maniator) [reply package.json:20]: Intentionally removed

File: package.json [hunk 3]
```diff
@@ -17,28 +17,31 @@
...
-        "integ:local:firefox": "npm run build:webpack:dev && cross-env NODE_OPTIONS=--max-http-header-size=80000 testcafe firefox --app \"http-server ./build/dev -s\"",
-        "integ:local:firefox:headless": "npm run build:webpack:dev && cross-env NODE_OPTIONS=--max-http-header-size=80000 testcafe firefox:headless --app \"http-server ./build/dev -s\"",
-        "integ:local:safari": "npm run build:webpack:dev && cross-env NODE_OPTIONS=--max-http-header-size=80000 testcafe safari --app \"http-server ./build/dev -s\"",
-        "integ:local:edge": "npm run build:webpack:dev && cross-env NODE_OPTIONS=--max-http-header-size=80000 testcafe edge --app \"http-server ./build/dev -s\"",
-        "integ:local:ie": "npm run build:webpack:dev && cross-env NODE_OPTIONS=--max-http-header-size=80000 testcafe ie --app \"http-server ./build/dev -s\" src/__integ__/unsupportedBrowser.test.ts",
+        "preinteg:browser": "npm run build:dev",
+        "integ:browser": "testcafe --config-file=./.testcafe.browser.js --app \"http-server ./build/dev -s\"",
```
Comment (qhanam) [inline package.json:41]: issue: The nice thing about having each browser as a separate script is that it is convenient for development. We could run testcafe directly, but it's convenient to simply call `npm run integ:local:chrome`.

Comment (maniator) [reply package.json:41]: forked from there, bit there is nothing preventing you from doing:

```npm run integ:browser -- chrome```

which will run tests in chrome :-)

Comment (qhanam) [reply package.json:41]: oh I see -- yep that seems like an elegant solution

**Review by qhanam — APPROVED** [review]

**Review by adebayor123 — COMMENTED** [review]
Hi @maniator would you be able to address the merge conflict? Once it's completed, I will go ahead and merge it.

**Comment by maniator** [comment]
> Hi @maniator would you be able to address the merge conflict? Once it's completed, I will go ahead and merge it.

@adebayor123 done! :-)

**Comment by adebayor123** [comment]
> > Hi @maniator would you be able to address the merge conflict? Once it's completed, I will go ahead and merge it.
> 
> @adebayor123 done! :-)

Another merge confict :( Could you resolve this one again, then I will merge it in.

**Comment by maniator** [comment]
> > > Hi @maniator would you be able to address the merge conflict? Once it's completed, I will go ahead and merge it.
> > 
> > 
> > @adebayor123 done! :-)
> 
> Another merge confict :( Could you resolve this one again, then I will merge it in.

oy... yes I can

**Comment by maniator** [comment]
@adebayor123 rebased yet again :-)

**Pull request merged**